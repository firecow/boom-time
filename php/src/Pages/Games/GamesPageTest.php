<?
declare(strict_types=1);

namespace App\Pages\Games;

use App\Exceptions\FileErrorException;
use App\Exceptions\FileNotFoundException;
use App\Mocks\ContextMock;
use App\Mocks\SuperGlobalsMock;
use App\Router\RouteHandler;
use App\Testing\RegressionTest;

class GamesPageTest extends RegressionTest
{

    /**
     * @throws FileErrorException
     * @throws FileNotFoundException
     */
    public function doTest(): string
    {
        $superGlobals = new SuperGlobalsMock();
        $superGlobals->setHTTPGetValue("pageKey", "/games/");
        $ctx = new ContextMock($superGlobals);
        $routeHandler = new RouteHandler($ctx);
        $routeResponse = $routeHandler->run("/rest/getpagehtml/");
        return $routeResponse->prettyPrint();
    }
}
