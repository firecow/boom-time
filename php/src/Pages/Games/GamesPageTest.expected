HTTP/1.1 200 Ok
Content-Length: 61
Content-Type: text/html

<div>
    <div>Mahjong</div>
    <div>Bubble Zoo</div>
</div>